### ISSUE TITLE
[pickers] Date Range Calendar doesn't select when using arrows with VoiceOver (Sequoia and Sonoma MacOS)

### ISSUE THREAD

**Issue by iseatz-zgoodwin** [mui/mui-x#15432]
### Steps to reproduce

Steps:
1. Open this link to live example: [MUI Date Range Calendar
](https://mui.com/x/react-date-pickers/date-range-calendar/)
2. Use Mac with Sonoma/Sequoia OS
3. Use Safari/Chrome/Edge/Firefox browsers
4. Navigate to a desired calendar to test (I tested single box calendar from section ["Customization"](https://mui.com/x/react-date-pickers/date-range-calendar/#customization))
5. TAB into calendar box (observe that first date is on focus)
6. Then use either arrow down/up/left/right or control+option+arrow down/up/left/right
7. Observe how arrows work (pay attention where they land and how focus works)
8. Press ENTER/SPACE
9. Observe how selection doesn't work

### Current behavior

We noticed that on our proprietary apps datepickers no longer properly work with VO but only on Sonoma and Seqouia MacOS (works fine on Ventura OS). The worst is on Firefox - arrows don't work at all. On Chrome/Edge the arrows work on the left/right but causing issues when using arrow up/down. On Safari arrows work all directions. But on all browsers ENTER/SPACE doesn't select the correct date - selection is always on what TAB landed on.
I additionally tested this datepicker with NVDA/JAWS 2024/JAWS 2025 on Windows 10 on Edge/Chrome/Firefox and there were no issues with those combinations.

Videos:
[Safari test](https://drive.google.com/file/d/1rARb037b9U2Xgkk19uj_WbExCQpCWheX/view?usp=sharing)

[Firefox test](https://drive.google.com/file/d/1ZsO65PSji3kQMON033tdpYjlMrjdG6uX/view?usp=sharing)

[Chrome test](https://drive.google.com/file/d/18VXkYNccX0K4wgwCPAwdSS0NRogZ5A5U/view?usp=sharing)

[Edge test](https://drive.google.com/file/d/14k6hSmRuoJqepsOqHPkhY_HaO-QqJ5t8/view?usp=sharing)

### Expected behavior

VoiceOver user should be able to change dates on the datepicker box using arrows and select chosen day with ENTER/SPACE.

### Context

It is hard to say if it's Apple issue or MUI issue, but it's a 50/50 possibility. However, this violates [WCAG 2.0/2.1/2.2 Level A SC 4.1.2 (Compatibility)](https://www.w3.org/WAI/WCAG21/Understanding/name-role-value.html) and puts this library users (companies) at risk of lawsuits.

### Your environment

Computer:
MacBook Pro, Apple M1 Max, 15.1 (24B83)

Browsers: 
Safari Version 18.1 (20619.2.8.11.10)
Chrome Version 130.0.6723.117 (Official Build) (arm64)
Edge Version 130.0.2849.80 (Official build) (arm64)
Firefox Version 132.0.2 (aarch64)

**Search keywords**: accessibility

**Search keywords**:

**Comment by michelengelen** [comment]
Hey @iseatz-zgoodwin and thanks for opening this issue ... 
> ... works fine on Ventura OS ...
this makes me think that this might be a MacOS related problem ... I'll ping the team though and see if we can provide a fix for this soon!

**Comment by LukasTy** [comment]
Hey @iseatz-zgoodwin
I've checked this case, looked into [usage documentation](https://mcmw.abilitynet.org.uk/how-to-use-voiceover-the-screen-reader-in-macos-15-sequoia) and can say that everything seems to be working correctly, at least on Safari.
Navigating with <kbd>Control</kbd>+<kbd>Option</kbd>+<kbd>Arrow</kbd> and then selecting with <kbd>Space</kbd> produces the expected result. 👍 

Other browsers do have some issues, but honestly, VO is not the industry standard for screen readers, I'm not sure if anyone is optimizing solutions to work best on [VO as compared to JAWS and NVDA](https://webaim.org/projects/screenreadersurvey10/).

Could you recheck if the behavior makes sense on your end given that only <kbd>Space</kbd> should select values?

**Comment by iseatz-zgoodwin** [comment]
@LukasTy 
Thank you for response, yes, I just tested with Safari and when using additionally Control + Option, then it works. However, on Chrome the Control + Option doesn't work at all (on desktop), I didn't test again on other 2 browsers, since indeed those combinations are rarer.
But now additionally tested on iPhone, and the good news it's possible to select with 2 taps, but only swiping left/right works, going up/down goes between week days letters and cancel/ok buttons. Which is minor inconvenience, I agree (but still a bug even if medium/low). And VO on phones is more popular than TalkBack per same screen reader studies.

I would still recommend to note these inconsistencies in your VPAT. I understand datepickers are one of the hardest custom widgets for accessibility.

**Comment by iseatz-zgoodwin** [comment]
@LukasTy apologies for late response. 
I tested Date Calendar on iPhone and left/right swipe goes linearly through all controls and elements (like dates in grid) and swiping up/down don't work on the dates grid, only on controllers for changing year, months. So basically, very similar issue like with arrows. However, it's still usable, not blocking experience, just slower experience.

**Comment by michelengelen** [comment]
@LukasTy gentle ping ^^

**Comment by LukasTy** [comment]
Placing the issue into grooming to improve the discovered deficiency when navigating on Mobile (Safari/iOS).
